#123 [Docs] AdnActivation 1.4.0 (#126)
Inbibo/adonisfx_docs: docs/maya/activation.md

# AdnActivation

> The AdnActivation node is a dependency node that allows operations on activation values for the computation of a final activation value that can be used to drive, for example, muscle activations. An example of its use case could be the merging of various sensor activations into one single driven activation to activate a muscle given multiple poses.

Reviewer: The description is perfect, honestly, but it sounded repetitive because of the "activation" occurrences. Just an idea:

```suggestion
The AdnActivation node is a dependency node that allows operations on a set of input values for the computation of a final value that can be used to drive, for example, muscle activations. An example of its use case could be the merging of the output value of various sensor together to activate a muscle given multiple poses.
```

Author: Good for me too. Thanks!

## How To Use

To create this node, follow these steps:

1. Open the Node Editor from Windows > Node Editor.
2. Press the Tab key, type *AdnActivation* and press Enter to create a new instance of this node.

The activation node can now be used to override, add, subtract, multiply or divide the activations from different sources (sensors) into one final activation value.
For example, multiple distance sensors of a character can be merged together to produce different kinds of activations throughout the simulation.

<figure markdown>
  ![AdnActivation example](images/adn_activation_create.png)
  <figcaption><b>Figure 1</b>: Use case in which AdnActivation is created for merging two activations to drive a muscle's activation.</figcaption>
</figure>

To add new inputs to the AdnActivation node:

> 1. Go to the AET and press *Add New Item*.

Reviewer: ```suggestion
1. Go to the Attribute Editor and press *Add New Item*.
```

Author: Done!

2. Set the desired *Operator*.
3. Type in a fixed value in *Value*, animated it or connect the plug with an AdonisFX sensor's output.

> [!NOTE]

> > All operators will be evaluated from top to bottom (starting from index 0 and ending on index N).

Reviewer: I like how you wrote it, but I beleive this is more user friendly

```suggestion
> All operators will be evaluated from top to bottom (starting from index 0 and ending on the last index used).
```

Author: Done!

## Operators

1. **Over (Override)**: Overrides the accumulated output activation value with Value. If the current accumulated activation value is 1.0 and Value is 2.0 the output activation will be 2.0.
2. **Add (Add)**: Adds the accumulated output activation value with Value. If the current accumulated activation value is 1.0 and Value is 2.0 the output activation will be 3.0.

> 3. **Sub (Subtract)**: Subtracts the accumulated output activation value with Value. If the current accumulated activation value is 1.0 and Value is 2.0 the output activation will be 0.0.

Reviewer: This really depends on the output min value. I think we should add rewrite this a little:
```suggestion
3. **Sub (Subtract)**: Subtracts the accumulated output activation value with Value. If the current accumulated activation value is 1.0 and Value is 2.0, then new accumulated values will be -1.0.
```

Using "the new accumulated" we are not referringto the final value. We will have to add another paragraph or a Note box saying: the final value will be clamped based on the min and max output values

Author: Done!

4. **Mult (Multiply)**: Multiplies the accumulated output activation value with Value. If the current accumulated activation value is 1.0 and Value is 2.0 the output activation will be 2.0.

> 5. **Div (Divide)**: Dives the accumulated output activation value with Value. If the current accumulated activation value is 1.0 and Value is 2.0 the output activation will be 0.5.

Reviewer 2: ```suggestion
5. **Div (Divide)**: Divides the accumulated output activation value with Value. If the current accumulated activation value is 1.0 and Value is 2.0 the output activation will be 0.5.
```

Author: Done!

## Example

<figure markdown>
  ![AdnActivation example setup](images/adn_activation_example.png)
  <figcaption><b>Figure 1</b>: Closeup use case in which AdnActivation is created for merging two activations to drive a muscle's activation.</figcaption>
</figure>

In the above setup we have the following characteristics:

> 1. 1 AdnSensorRotation.

Reviewer: To make it easer to read
```suggestion
1. One AdnSensorRotation.
```

Author: Done!

2. 1 AdnSensorPosition.
3. 1 AdnActivation.
4. 1 AdnMuscle.
5. 2 Inputs added to AdnActivation.
6. Input 1 has an Over operator and is connected to the AdnSensorRotation.
7. Input 2 has an Add operator and is connected to the AdnSensorPosition.
8. The bypass option is unchecked for both inputs.
9. Out Value will be: AdnSensorRotation Activation + AdnSensorPosition Activation.

## Attributes

### Inputs Attributes
The *inputs* attribute is presented as an array of 3 attributes which can be found below.

| Name | Type | Default | Animatable | Description |
| :--- | :--- | :------ | :--------- | :---------- |

> | **Bypass Operator** | Boolean | True            | ✓ | If enabled, it bypasses the current operator in the input list, which will not contribute to the final activation value |

Reviewer 2: ```suggestion
| **Bypass Operator** | Boolean | True            | ✓ | If enabled, it bypasses the current operator in the input list, which will not contribute to the final activation value. |
```

Author: Done!

| **Value**           | Float   | 0.0             | ✓ | Activation value that will contribute, given the operator type, to the final activation. |
| **Operator**        | Enum    | 0 (Over)        | ✓ | Operator used to contribute to the final activation. This can be: (0) Over; (1) Add; (2) Sub; (3) Mult; (4) Divide. |

### Output Attributes
| Name | Type | Default | Animatable | Description |
| :--- | :--- | :------ | :--------- | :---------- |
| **Output Min** | Float | 0.0 | ✓ | Minimum supported output activation value. |

> | **Output Max** | Float | 1.0 | ✓ | Maximum supported output activation value. |

Reviewer: Could you add the range (if any), I don't remember

Author: Added. I did not add them because they are soft. But I've added a clarification
